mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala

```diff
@@ -438,6 +438,10 @@ private[ml] object SummaryBuilderImpl extends Logging {
         while (i < len) {
           realVariance(i) = (currM2n(i) + deltaMean(i) * deltaMean(i) * weightSum(i) *
             (totalWeightSum - weightSum(i)) / totalWeightSum) / denominator
+          // Because of numerical error, it is possible to get negative real variance
+          if (realVariance(i) < 0.0) {
```

Comment:
The computation of _variance_ may be touch this numerical error, it seems ```WeightedLeastSquares``` also use the same method to compute _variance_ , does it will have similar issue? @WeichenXu123